src/test/java/com/artipie/api/ManageUsersTest.java

```diff
@@ -0,0 +1,182 @@
+            true
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void returnsEmptyList(final boolean add) {
+        if (add) {
+            this.blsto.save(ManageUsersTest.KEY, new byte[]{});
+        }
+        MatcherAssert.assertThat(
+            this.users.list(),
+            new IsEmptyCollection<>()
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("creds")
+    void addsUser(final Pair<YamlMapping, Boolean> pair) {
+        if (pair.getRight()) {
+            this.blsto.save(
+                ManageUsersTest.KEY,
+                pair.getLeft().toString().getBytes(StandardCharsets.UTF_8)
+            );
+        }
+        final String alice = "Alice";
+        final String email = "Alice@example.com";
+        final String pass = "xyz";
+        this.users.add(
+            Json.createObjectBuilder().add("type", "plain").add("pass", pass)
+                .add("email", email)
+                .add("groups", Json.createArrayBuilder().add("reader").add("creator").build())
+                .build(),
+            alice
+        );
+        final JsonArray list = this.users.list();
+        final JsonObject nuser = list.stream()
+            .filter(usr -> alice.equals(usr.asJsonObject().getString("name")))
+            .findFirst().get().asJsonObject();
+        MatcherAssert.assertThat(
+            "Failed to add user email",
+            nuser.getString("email"),
+            new IsEqual<>(email)
+        );
+        MatcherAssert.assertThat(
+            "Failed to add password",
+            nuser.getString("pass").toString().equals(pass)
+        );
+        MatcherAssert.assertThat(
+            "Failed to add groups",
+            nuser.get("groups").asJsonArray().stream().map(JsonValue::toString)
+                .map(item -> item.replace("\"", "")).collect(Collectors.toList()),
+            Matchers.containsInAnyOrder("reader", "creator")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("creds")
+    void throwsErrorWhenUserNotExist(final Pair<YamlMapping, Boolean> pair) {
+        if (pair.getRight()) {
+            this.blsto.save(
+                ManageUsersTest.KEY,
+                pair.getLeft().toString().getBytes(StandardCharsets.UTF_8)
+            );
+        }
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            () -> this.users.remove("Alice")
+        );
+    }
+
+    @Test
+    void removesUser() {
+        this.blsto.save(
+            ManageUsersTest.KEY,
+            new CredsConfigYaml().withUsers("John", "Mark").yaml().toString()
+                .getBytes(StandardCharsets.UTF_8)
+        );
+        this.users.remove("John");
+        MatcherAssert.assertThat(
+            this.users.list().size(),
+            new IsEqual<>(1)
+        );
+    }
+
```

Comment:
Please add negative scenario for case removing of non existing user.

Reply:
check line 140 please, there is such test

Reply:
ok, thanks